api/plugins/session-auth.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  DoneFuncWithErrOrRes,
+  FastifyPluginCallback,
+  FastifyReply,
+  FastifyRequest
+} from 'fastify';
+import fp from 'fastify-plugin';
+
+const sessionAuth: FastifyPluginCallback = (fastify, _opts, done) => {
+  fastify.decorate('authenticateSession', authenticateSession);
```

Comment:
If we use a const (see below), `decorate` will need to be called after that's declared.